tests/console/check_system_info.pm

```diff
@@ -91,11 +91,13 @@ sub check_nodejs_common {
     record_info('SLE-21783', 'check nodejs-common');
     zypper_call('in nodejs-common | tee /tmp/install_nodejs.log');
 
-    assert_script_run('grep -E "Installing: nodejs16" /tmp/install_nodejs.log');
-    if ((script_output(q(zypper se -i nodejs | sed -n '/nodejs16 /p' | awk '{print $1}')) !~ /i/) || \
-        (script_output(q(zypper se -i nodejs | sed -n '/nodejs-common/p' | awk '{print $1}')) !~ /i\+/)) {
-        die "Expected package is not installed";
+    my $ret = script_run(q(grep "'nodejs-common' is already installed." /tmp/install_nodejs.log));
+    if ($ret != 0) {
+        assert_script_run('grep -E "Installing: nodejs16" /tmp/install_nodejs.log');
     }
+    my $ret_s1 = script_output(q(zypper se -i nodejs | sed -n '/nodejs16 /p' | awk '{print $1}'));
+    my $ret_s2 = script_output(q(zypper se -i nodejs | sed -n '/nodejs-common/p' | awk '{print $1}'));
+    die "Expected package is not installed" if ('i' ne "$ret_s1" || 'i+' ne "$ret_s2");
```

Comment:
It is hard to say which is better to compare with the old code.

Reply:
and the old code can't work
https://openqa.nue.suse.com/tests/8502837#step/check_system_info/39